apps/homescreen/js/app.js

```diff
@@ -726,6 +757,100 @@
       return this.icons.children.indexOf(child);
     },
 
+    removeSelectedIcon: function() {
+      if (!this.selectedIcon) {
+        return;
+      }
+
+      if (this.selectedIcon.app) {
+        if (!this.selectedIcon.app.removable) {
+          return;
+        }
+        navigator.mozApps.mgmt.uninstall(this.selectedIcon.app);
+      } else if (this.selectedIcon.bookmark) {
+        new MozActivity({
+          name: 'remove-bookmark',
+          data: { type: 'url', url: this.selectedIcon.bookmark.id }
+        });
+      }
+    },
+
+    renameSelectedIcon: function() {
+      if (!this.selectedIcon || !this.selectedIcon.bookmark) {
+        return;
+      }
+
+      new MozActivity({
+        name: 'save-bookmark',
+        data: { type: 'url', url: this.selectedIcon.bookmark.id }
+      });
+    },
+
+    iconIsEditable: function(icon) {
+      return (icon.bookmark || (icon.app && icon.app.removable)) ? true : false;
```

Comment:
Because that ends up returning the objects (you could put a !! before it, which would be similar to what I've done). Most things are fine, but it came up in a unit test and I've had funny things happen from returning things like `undefined` instead of `false`, so decided to be explicit.